src/routes/books/books.controller.js

```diff
@@ -0,0 +1,145 @@
+const fs = require("fs");
+const { promisify } = require("util");
+const writeFile = promisify(fs.writeFile);
+const { validationResult } = require("express-validator");
+
+const { ErrorHandler } = require("../../utils/errors")
+const { getErrorMessage } = require('./utils')
+const { STATUS_CODES } = require('../../constants')
+const booksData = require("../../../db/books.json");
+
+const deleteBooks = async (req, res) => {
+    await writeFile("db/books.json", JSON.stringify([]));
+
+    return res
+        .status(STATUS_CODES["OK"])
+        .json({
+            data: booksData,
+        })
+}
+
+const deleteBooksById = async (req, res, next) => {
+    const { bookId } = req.params;
+    const bookIndex = booksData.findIndex(({ bookID }) => bookID === bookId)
+
+    if (bookIndex === -1) {
+        return res
+            .status(STATUS_CODES["NOT_FOUND"])
+            .json({
+                errors: [{
+                    msg: getErrorMessage(bookId)
+                }]
+            });
+    }
+
+    const updatedBooksData = [...booksData]
+    const deletedBook = updatedBooksData.splice(bookIndex, 1);
```

Comment:
```suggestion
    const deletedBook = booksData.splice(bookIndex, 1);
```
We can potentially do it in one line since `booksData` gets read every time this file is loaded anyway; but nothing wrong with what you had here.